### PR TITLE
Batching: Add Protobuff params to message Function for batching

### PR DESCRIPTION
**Pull request by cathyzbn**
## Describe your changes
Protoc changes to to support batching

---

Check these boxes or delete any item (or this section) if not relevant for this PR.

- [x] Client+Server: this change is compatible with old servers
- [x] Client forward compatibility: this change ensures client can accept data intended for later versions of itself

Note on protobuf: protobuf message changes in one place may have impact to
multiple entities (client, server, worker, database). See points above.

### CODE REVIEW

File: modal/functions.py [hunk 1]
```diff
@@ -809,6 +811,8 @@ async def _load(self: _Function, resolver: Resolver, existing_object_id: Optiona
                     object_dependencies=object_dependencies,
                     block_network=block_network,
                     max_inputs=max_inputs or 0,
+                    max_num_batch_inputs=max_num_batch_inputs or 1,
```
Comment (gongy) [inline modal/functions.py:814]: Internally, I think we should use `0` to mean "no batching", right? so `or 0`

File: modal/functions.py [hunk 2]
```diff
@@ -809,6 +811,8 @@ async def _load(self: _Function, resolver: Resolver, existing_object_id: Optiona
...
+                    max_batch_timout_sec=max_batch_timout_sec or 0,
```
[the remark this answers is not in the document]
Comment (cathyzbn) [reply modal/functions.py:815]: my bad!

File: modal/functions.py [hunk 3]
```diff
@@ -496,6 +496,8 @@ def from_args(
         allow_background_volume_commits: Optional[bool] = None,
         block_network: bool = False,
         max_inputs: Optional[int] = None,
+        batch_max_num_inputs: Optional[int] = None,
+        batch_max_timeout_sec: Optional[float] = None,
```
Comment (mwaskom) [inline modal/functions.py:500]: Super nit, but can we use `_secs` to be consistent with other fields?

```
$ git grep "_sec[s ]" modal_proto/
modal_proto/api.proto:  uint32 timeout_secs = 21;
modal_proto/api.proto:  uint32 task_idle_timeout_secs = 25;
modal_proto/api.proto:  optional uint32 timeout_secs = 6;
modal_proto/api.proto:  optional uint32 task_idle_timeout_secs = 7;
modal_proto/api.proto:  uint32 timeout_secs = 7;
```

Comment (mwaskom) [reply modal/functions.py:500]: Also, while I'm being really annoying, "max timeout seconds" doesn't make sense IMO.

The concept of a "max" waiting period is already included in the "timeout" term. But reaching a "timeout" is usually bad / an error. So I would say something like `max_wait_secs` is a better name here.

Comment (cathyzbn) [reply modal/functions.py]: that makes sense! sounds good!

**Pull request closed without merging**